src/App.tsx

```diff
@@ -15,36 +16,94 @@
   'Garlic',
 ];
 
+const SORT_FIELD_ALPHABETICCALY = 'alphabeticcaly';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  reverse: boolean,
+) {
+  let newVisibleGoods = [...goods];
+
+  if (sortField === SORT_FIELD_ALPHABETICCALY) {
+    newVisibleGoods.sort();
+  }
+
+  if (sortField === SORT_FIELD_LENGTH) {
+    newVisibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (reverse) {
+    return newVisibleGoods.reverse();
+  }
+
+  if (sortField === '') {
+    newVisibleGoods = [...goods];
+  }
+
+  return newVisibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverce] = useState(false);
```

Comment:
There is a typo in the state setter function name `setReverce`. It should be `setReverse` to match the state variable name `reverse`.